src/components/Pagination/Pagination.component.tsx

```diff
@@ -0,0 +1,141 @@
+import type { FC, JSX } from 'react';
+
+import { ArrowLeft } from '@/assets/icons/ArrowLeft.tsx';
+import { ArrowRight } from '@/assets/icons/ArrowRight.tsx';
+import { PaginationButton } from '@/components/PaginationButton/PaginationButton.component.tsx';
+import type { Themes } from '@/types/themes.type.ts';
+
+import styles from './Pagination.module.css';
+
+interface PaginationProps {
+    totalItems: number;
+    currentPage: number;
+    handlePageChange: (page: number) => void;
+    theme: Themes;
+    totalPages: number;
+}
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+export const Pagination: FC<PaginationProps> = ({ totalItems, theme, currentPage, handlePageChange, totalPages }) => {
+    const firstPage = 1;
+
+    const isOnFirstPage = currentPage === firstPage;
+    const isOnLastPage = currentPage === totalPages;
+
+    const goToNextPage = () => {
+        if (!isOnLastPage) {
+            handlePageChange(currentPage + 1);
+        }
+    };
+
+    const goToPreviousPage = () => {
+        if (!isOnFirstPage) {
+            handlePageChange(currentPage - 1);
+        }
+    };
+
+    const paginationButtons: JSX.Element[] = [];
```

Comment:
Your component looks good and do it job, and I recommend you to move your `paginationButtons` into separate helper function `getPaginationButtons` (naming is up to you) that will improve the readability and debugging

Your components should be clean as possible